src/os/shared.rs

```diff
@@ -34,17 +34,16 @@ impl Iterator for KeyboardEvents {
 
 fn get_datalink_channel(
     interface: &NetworkInterface,
-) -> Result<Box<dyn DataLinkReceiver>, std::io::Error> {
+) -> Result<Box<dyn DataLinkReceiver>, GetInterfaceError> {
     let mut config = Config::default();
     config.read_timeout = Some(time::Duration::new(1, 0));
-
     match datalink::channel(interface, config) {
         Ok(Ethernet(_tx, rx)) => Ok(rx),
-        Ok(_) => Err(std::io::Error::new(
-            ErrorKind::Other,
-            "Unsupported interface type",
-        )),
-        Err(e) => Err(e),
+        Ok(_) => {
+            let error = GetInterfaceError::new(GetInterfaceErrorKind::OtherError("Please do something".to_string()));
+            Err(error)
+        },
+        Err(e) => Err(GetInterfaceError::new(GetInterfaceErrorKind::PermissionError("Permission error".to_string()))),
```

Comment:
Permission error is only one error type we can get here. Essentially, we would like to forward all of these errors to the user, and if the error is a permission error, display the "use sudo" string below. Does this make sense?